webapp/src/Service/Compare/ScoreboardCompareService.php

```diff
@@ -0,0 +1,134 @@
+                        $problemForSecond = $problem2;
+                        break;
+                    } elseif (preg_match('/^Id = ' . preg_quote($problem->problemId, '/') . '-+\d+$/', $problem2->problemId)) {
+                        $problemForSecond = $problem2;
+                        break;
+                    }
+                }
+
+                if ($problemForSecond === null && $problem->solved) {
+                    $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem %s solved in first file, but not found in second file', $index, $problem->problemId));
+                } elseif ($problemForSecond !== null && $problem->solved !== $problemForSecond->solved) {
+                    $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem %s solved does not match', $index, $problem->problemId), (string)$problem->solved, (string)$problemForSecond->solved);
+                }
+
+                if ($problemForSecond) {
+                    if ($problem->numJudged !== $problemForSecond->numJudged) {
+                        $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem %s num judged does not match', $index, $problem->problemId), (string)$problem->numJudged, (string)$problemForSecond->numJudged);
+                    }
+
+                    if ($problem->numPending !== $problemForSecond->numPending) {
+                        $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem %s num pending does not match', $index, $problem->problemId), (string)$problem->numPending, (string)$problemForSecond->numPending);
+                    }
+
+                    if ($problem->time !== $problemForSecond->time) {
+                        // This is an info message for now, since PC^2 doesn't expose time info
+                        $this->addMessage(MessageType::INFO, sprintf('Row %d: Problem %s time does not match', $index, $problem->problemId), (string)$problem->time, (string)$problemForSecond->time);
+                    }
+                }
+            }
+
+            foreach ($object2->rows[$index]->problems as $problem2) {
+                $problemForFirst = null;
+
+                foreach ($row->problems as $problem) {
+                    // PC^2 uses different problem ID's. For now also match on `Id = {problemId}-{digits}`
+                    if ($problem->problemId === $problem2->problemId) {
+                        $problemForFirst = $problem;
+                        break;
+                    } elseif (preg_match('/^Id = ' . preg_quote($problem->problemId, '/') . '-+\d+$/', $problem2->problemId)) {
+                        $problemForFirst = $problem;
+                        break;
+                    }
+                }
+
+                if ($problemForFirst === null && $problem2->solved) {
+                    $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem %s solved in second file, but not found in first file', $index, $problem2->problemId));
```

Comment:
```suggestion
                    $this->addMessage(MessageType::ERROR, sprintf('Row %d: Problem `%s` solved in second file, but not found in first file', $index, $problem2->problemId));
```
Makes it easier to read IMO.